src/clj/game/cards/upgrades.clj

```diff
@@ -20,6 +20,25 @@
                                  :prompt "Rez ICE with rez cost lowered by 3?" :priority 2
                                  :yes-ability {:effect (effect (rez-cost-bonus -3) (rez target))}}}}}
 
+   "Arella Salvatore"
+   {:events
+    {:agenda-scored
+     {:req (req (= (:previous-zone target) (:zone card)))
+      :interactive (req true)
+      :optional {:prompt "Install a card from HQ?"
```

Comment:
Similar to #3723, I prefer optional-then-card-selection prompts to be a single selection prompt, with the `:cancel-effect` being used as the `:no-ability` equivalent when needed. The prompt can then read "Install a card from HQ with Arella Salvatore?".